src/sentry/workflow_engine/metric_helpers.py

```diff
@@ -0,0 +1,166 @@
+from sentry.incidents.grouptype import MetricAlertFire
+from sentry.incidents.models.alert_rule import (
+    AlertRule,
+    AlertRuleThresholdType,
+    AlertRuleTrigger,
+    AlertRuleTriggerAction,
+)
+from sentry.snuba.models import QuerySubscription
+from sentry.users.services.user import RpcUser
+from sentry.workflow_engine.models import (
+    Action,
+    AlertRuleDetector,
+    AlertRuleTriggerDataCondition,
+    AlertRuleWorkflow,
+    DataCondition,
+    DataConditionGroup,
+    DataConditionGroupAction,
+    DataSource,
+    Detector,
+    DetectorState,
+    DetectorWorkflow,
+    Workflow,
+    WorkflowDataConditionGroup,
+)
+from sentry.workflow_engine.models.data_condition import Condition, ConditionType
+from sentry.workflow_engine.types import ActionType, DataSourceType, DetectorPriorityLevel
+
+
+def create_metric_action(alert_rule_trigger_action: AlertRuleTriggerAction) -> None:
+    try:
+        alert_rule_trigger_data_condition = AlertRuleTriggerDataCondition.objects.get(
+            alert_rule_trigger=alert_rule_trigger_action.alert_rule_trigger
+        )
+    except AlertRuleTriggerDataCondition.DoesNotExist:
+        return None
+
+    data = {
+        "type": alert_rule_trigger_action.type,
+        "sentry_app_id": alert_rule_trigger_action.sentry_app_id,
+        "sentry_app_config": alert_rule_trigger_action.sentry_app_config,
+    }
+    action = Action.objects.create(
+        required=False,
+        type=ActionType.NOTIFICATION,
+        data=data,
+        integration_id=alert_rule_trigger_action.integration_id,
+        target_display=alert_rule_trigger_action.target_display,
+        target_identifier=alert_rule_trigger_action.target_identifier,
+        target_type=alert_rule_trigger_action.target_type,
+    )
+    DataConditionGroupAction.objects.create(
+        condition_group_id=alert_rule_trigger_data_condition.data_condition.condition_group.id,
+        action_id=action.id,
+    )
+
+
+def create_metric_data_condition(alert_rule_trigger: AlertRuleTrigger) -> None:
+    try:
+        alert_rule_detector = AlertRuleDetector.objects.get(
+            alert_rule=alert_rule_trigger.alert_rule
+        )
+    except AlertRuleDetector.DoesNotExist:
+        return None
+
+    data_condition_group = alert_rule_detector.detector.workflow_condition_group
+    if not data_condition_group:
+        return None
+
+    condition_result = (
+        DetectorPriorityLevel.MEDIUM
+        if alert_rule_trigger.label == "warning"
+        else DetectorPriorityLevel.HIGH
+    )
+    threshold_type = alert_rule_trigger.alert_rule.threshold_type
+    condition = (
+        Condition.GREATER if threshold_type == AlertRuleThresholdType.ABOVE else Condition.LESS
```

Comment:
For UI purposes, do we need to include a case here for `AlertRuleThresholdType.ABOVE_AND_BELOW`?

Reply:
Yes, good catch. I should reach out to design too and make sure the anomaly detection UI includes this (and anything else specific to it).

Reply:
I ended up leaving a note to add this for anomaly detection later on since adding `Condition.ABOVE_OR_BELOW` includes creating the condition handler for anomaly detection which is a separate piece of work.